tests/ashley/machina_extensions/forum/test_forum_view.py

```diff
@@ -0,0 +1,411 @@
+        url = f"{self.url_list_topic}?o=DATE.0.7"
+        # Run
+        response = self.client.get(url)
+        # Check
+        assert response.status_code == 200
+
+        # Should have 1 sorted column as the other two are unknown format
+        self.assertContains(response, "sorted ", count=1)
+
+        # it should be in alphabetic order
+        self.assertContentBefore(response, "TOPIC A", "TOPIC B")
+        self.assertContentBefore(response, "TOPIC B", "TOPIC C")
+
+    def test_sorting_context_wisiwig(self):
+        """Test the relation between our context header and links created in the view
+        to ensure it's used
+        """
+        # sort on three fields
+        url = f"{self.url_list_topic}?o=-2.0.3"
+        # Run
+        response = self.client.get(url)
+
+        self.assertEqual(
+            response.context["header"],
+            TopicSortHeaderFactory.build_column_order(
+                [
+                    {
+                        "sorted": True,
+                        "ascending": True,
+                        "sort_priority": 2,
+                        "url_primary": "?o=-0.-2.3",
+                        "url_remove": "?o=-2.3",
+                        "url_toggle": "?o=-2.-0.3",
+                        "class_attrib": "sorted ascending",
+                    },
+                    {
+                        "url_primary": "?o=1.-2.0.3",
+                        "url_remove": "?o=-2.0.3",
+                        "url_toggle": "?o=-2.0.3",
+                    },
+                    {
+                        "sorted": True,
+                        "ascending": False,
+                        "sort_priority": 1,
+                        "url_primary": "?o=2.0.3",
+                        "url_remove": "?o=0.3",
+                        "url_toggle": "?o=2.0.3",
+                        "class_attrib": "sorted descending",
+                    },
+                    {
+                        "sorted": True,
+                        "ascending": True,
+                        "sort_priority": 3,
+                        "url_primary": "?o=-3.-2.0",
+                        "url_remove": "?o=-2.0",
+                        "url_toggle": "?o=-2.0.-3",
+                        "class_attrib": "sorted ascending",
+                    },
+                ],
+            ),
+        )
+
+        # Check topics links contains generated links
+        self.assertContains(
+            response,
+            (
+                '<a href="?o=-0.-2.3">Topics </a>'
+                '<span class="sortoptions">'
+                '<a class="sortremove" href="?o=-2.3" title="Remove from sorting"></a>'
+                '<span class="sortpriority" title="Sorting priority: 2">2</span>'
+                '<a href="?o=-2.-0.3" class="toggle ascending" title="Toggle sorting"></a>'
+                "</span>"
+            ),
+            html=True,
+        )
```

Comment:
Ok I assume that templates are tested here :slightly_smiling_face: 
You can ignore my comment on the fact that you test the response context, then ;)

Reply:
ok then great !